src/app/messages/messages.component.html

```diff
@@ -0,0 +1,8 @@
+<div *ngIf="messageService.messages.length">
+
+  <h2>Messages</h2>
+  <button class="clear"
+          (click)="messageService.clear()">clear</button>
```

Comment:
Du greifst hier auf einen Message zu, der als "private" deklariert wurde: `private messageService: MessageService`

Bitte nicht machen. Erstelle lieber eine `clear` Methode in der Component die dann das clear triggert. Code-Separation und so.

Reply:
Da habe ich nich vertan. Das messageService soll public sein. Hab nochmal im Tutorial nachgeguckt....